core-domain/src/test/java/org/uniprot/core/uniprotkb/description/impl/ProteinNameImplTest.java

```diff
@@ -103,20 +99,70 @@ void testOnlyFullAndShort() {
 
     @Test
     void needDefaultConstructorForJsonDeserialization() {
-        ProteinRecName obj = new ProteinRecNameImpl();
+        ProteinName obj = new ProteinNameImpl();
         assertNotNull(obj);
         assertFalse(obj.hasFullName());
     }
 
     @Test
     void builderFrom_constructorImp_shouldCreate_equalObject() {
-        ProteinRecName impl =
-                new ProteinRecNameImpl(
+        ProteinName impl =
+                new ProteinNameImpl(
                         new NameImpl(), Collections.emptyList(), Collections.emptyList());
-        ProteinRecName obj = ProteinRecNameBuilder.from(impl).build();
+        ProteinName obj = ProteinNameBuilder.from(impl).build();
 
         assertTrue(impl.hasFullName());
         assertTrue(impl.equals(obj) && obj.equals(impl));
         assertEquals(impl.hashCode(), obj.hashCode());
     }
+
+    public static ProteinName createProteinRecName(int listSize, boolean includeEvidences) {
```

Comment:
This method (createProteinRecName) and the below one (createProteinAltName) are identical, do we need to have both? Maybe a createProteinName would be enough.

Reply:
I think the other methods below can be optimised as well.. right?

Reply:
done